src/Storages/System/StorageSystemConfigs.cpp

```diff
@@ -0,0 +1,90 @@
+#include <boost/algorithm/string/join.hpp>
+#include <Poco/Util/AbstractConfiguration.h>
+
+#include <Columns/ColumnsNumber.h>
+#include <Processors/Sources/SourceFromSingleChunk.h>
+#include <QueryPipeline/Pipe.h>
+#include <Interpreters/Context.h>
+#include <Common/JSONBuilder.h>
+#include <Formats/FormatFactory.h>
+#include <IO/ReadBufferFromString.h>
+
+#include "StorageSystemConfigs.h"
+
+namespace DB
+{
+
+StorageSystemConfigs::StorageSystemConfigs(const StorageID & table_id_) : IStorage(table_id_)
+{
+    StorageInMemoryMetadata storage_metadata;
+    storage_metadata.setColumns(ColumnsDescription{names_and_types_list});
+    setInMemoryMetadata(storage_metadata);
+}
+
+static JSONBuilder::ItemPtr configToJSON(
+    const Poco::Util::AbstractConfiguration & config, const String & prefix = "")
+{
+    std::vector<String> keys;
+    config.keys(prefix, keys);
+    if (keys.empty())
+        return std::make_unique<JSONBuilder::JSONString>(config.getString(prefix));
+
+    std::unordered_map<String, std::vector<String>> keys_map;
+    for (const auto & key : keys)
+    {
+        auto pos = key.find_first_of('[');
+        auto obj_key = pos == std::string::npos ? key : key.substr(0, pos);
+        if (obj_key == "system" || obj_key == "daemon" || obj_key == "applicatioin")
+            continue;
+        keys_map[obj_key].push_back(key);
+    }
+
+    auto map = std::make_unique<JSONBuilder::JSONMap>();
+    for (const auto & [obj_key, keys_list] : keys_map)
+    {
+        if (keys_list.size() == 1)
+        {
+            map->add(obj_key, configToJSON(config, prefix + (prefix.empty() ? "" : ".") + keys_list[0]));
+        }
+        else
+        {
+            auto array = std::make_unique<JSONBuilder::JSONArray>();
+            for (const auto & key : keys_list)
+            {
+                array->add(configToJSON(config, prefix + (prefix.empty() ? "" : ".") + key));
+            }
+            map->add(obj_key, std::move(array));
+        }
+    }
+    return std::move(map);
+}
+
+Pipe StorageSystemConfigs::read(
+    const Names & column_names,
+    const StorageSnapshotPtr & storage_snapshot,
+    SelectQueryInfo &,
+    ContextPtr context,
+    QueryProcessingStage::Enum /*processed_stage*/,
+    const size_t /*max_block_size*/,
+    const unsigned /*num_streams*/)
+{
+    storage_snapshot->check(column_names);
```

Comment:
This table should not be allowed to read by regular user, so you should put something like:

```c
getContext()->checkAccess(AccessType::SYSTEM_CONFIGS);
```

And add this new type into `src/Access/Common/AccessType.h `

Also it worth adding a test, and cover:
- some config directive
- and that this privilege is required

Reply:
Thank you for your advice. It will be done.

Reply:
done